### PR TITLE
Support resource level rule replacement in hotspot module

### PR DESCRIPTION
**Pull request by luckyxiaoqiang**
### Describe what this PR does / why we need it
Support resource level rule replacement in hotspot module.

### Special notes for reviews

### CODE REVIEW

File: core/hotspot/rule_manager.go [hunk 1]
```diff
@@ -137,7 +146,13 @@ func ClearRules() error {
 	return err
 }
 
-func onRuleUpdate(rules []*Rule) (err error) {
+// ClearRulesOfResource clears resource level parameter flow rules.
```
Comment (liqiangz) [inline core/hotspot/rule_manager.go:149]: flow rules -> hotspot rules

File: core/hotspot/rule_manager.go [hunk 2]
```diff
@@ -237,14 +210,98 @@ func onRuleUpdate(rules []*Rule) (err error) {
 
 	tcMux.Lock()
 	tcMap = m
-	currentRules = rules
+	currentRules = rawResRulesMap
```
Comment (liqiangz) [inline core/hotspot/rule_manager.go:213]: We can do this without lock.

File: core/hotspot/rule_manager.go [hunk 3]
```diff
@@ -237,14 +210,98 @@ func onRuleUpdate(rules []*Rule) (err error) {
...
+
+	tcMux.Lock()
+	if len(newResTcs) == 0 {
+		delete(tcMap, res)
+	} else {
+		tcMap[res] = newResTcs
+	}
+	tcMux.Unlock()
+	currentRules[res] = rawResRules
+	logging.Debug("[HotSpot onResourceRuleUpdate] Time statistic(ns) for updating flow rule", "timeCost", util.CurrentTimeNano()-start)
```
Comment (liqiangz) [inline core/hotspot/rule_manager.go:267]: flow rules -> hotspot rules

File: core/hotspot/rule_manager.go (hunk 1 above)
Comment (luckyxiaoqiang) [reply core/hotspot/rule_manager.go:149]: `parameter flow rules` is the history name, it ` means `hotspot rules`.

File: core/hotspot/rule_manager.go (hunk 3 above)
Comment (luckyxiaoqiang) [reply core/hotspot/rule_manager.go]: Updated.

File: core/hotspot/rule_manager.go (hunk 2 above)
Comment (luckyxiaoqiang) [reply core/hotspot/rule_manager.go]: Updated.

**Review by sanxun0325 — APPROVED** [review]
LGTM

**Review by liqiangz — APPROVED** [review]
LGTM

**Review by louyuting — APPROVED** [review]

**Pull request merged**